### PR TITLE
Restructuring

### PR DESCRIPTION
**Pull request by jalavik**
Fully makes the plotextractor utility a standalone library with new features:

* Removes integration with Invenio and the CLI.

* Avoids usage of shell commands and instead uses ImageMagick bindings (Wand) for converting and rotating images

* Removes MARCXML specific output and rather outputs Python dictionaries.

* Adds tests and skeleton docs.

@kaplun

### CODE REVIEW

**Comment by kaplun** [comment]
Stupid question first: have you based this PR on the very latest cookiecutter improvements?

**Comment by jalavik** [comment]
Partly yes, but not fully, as it is not an inveno-cookiecutter module but a standalone lib. Still it should be mostly up to date. Minor things changes all the time, but docs building, testing and the important stuff should be okay.

File: plotextractor/converter.py [hunk 1]
```diff
@@ -22,45 +22,42 @@
...
+        raise InvalidTarball
 
-    tex_file_extension = 'tex'
-    image_output_contains = 'image'
-    eps_output_contains = '- type eps'
-    ps_output_contains = 'Postscript'
+    tarball = tarfile.open(original_tarball)
+    if not output_directory:
+        # No directory given, so we use the same directory as the tarball
+        output_directory = os.path.dirname(original_tarball)
```
Comment (kaplun) [inline plotextractor/converter.py]: This is not safe. What if the `original_tarball` contains paths that already exist in the current directory? Shall we rather default to a tempdir named after `original_tarball` if what we want to do is to have the unpacked output next to the file?

File: plotextractor/converter.py [hunk 2]
```diff
@@ -161,14 +150,9 @@ def convert_images(image_list):
...
-        # conversions; (2) the arguments should be passed properly so
-        # that they are escaped.
-
-        dummy1, cmd_out, dummy2 = run_shell_command('file %s', (image_file,))
+        cmd_out = check_output(['file', image_file], timeout=timeout)
         if cmd_out.find(png_output_contains) > -1:
+            # Already PNG
             ret_list.append(image_file)
         else:
             # we're just going to assume that ImageMagick can convert all
```
Comment (kaplun) [inline plotextractor/converter.py]: Time to use pillow :dancer:  (OK maybe in a subsequent PR)

File: plotextractor/converter.py [hunk 3]
```diff
@@ -176,33 +160,47 @@ def convert_images(image_list):
...
-                dummy1, cmd_out, cmd_err = run_process_with_timeout(cmd_list)
-                if cmd_err == '' or os.path.exists(converted_image_file):
-                    ret_list.append(converted_image_file)
-                else:
-                    write_message('convert failed on ' + image_file)
-            except Timeout:
-                write_message('convert timed out on ' + image_file)
+            convert_image(image_file, converted_image_file, image_format)
+            if os.path.exists(converted_image_file):
+                ret_list.append(converted_image_file)
```
Comment (kaplun) [inline plotextractor/converter.py]: Are we also preserving the original vector format?

File: plotextractor/converter.py (hunk 1 above)
Comment (jalavik) [inline plotextractor/converter.py]: Indeed. This was leftover. Good catch. It is handled better in `api.py:process_tarball`.

File: plotextractor/converter.py (hunk 3 above)
Comment (jalavik) [inline plotextractor/converter.py]: That's a good point. We aren't but we'd want to. I will try to quickly add it.

Comment (kaplun) [inline plotextractor/converter.py]: :kissing: That would be amazing and solving a long standing issue.

Comment (kaplun) [inline plotextractor/converter.py]: https://github.com/inveniosoftware/invenio/issues/316

File: plotextractor/converter.py (hunk 1 above)
Comment (kaplun) [inline plotextractor/converter.py]: At least be sure the checkmanifest trick is in place, which can save your day when making a release later on pypi

**Comment by kaplun** [comment]
> Partly yes, but not fully, as it is not an inveno-cookiecutter module but a standalone lib. Still it should be mostly up to date. Minor things changes all the time, but docs building, testing and the important stuff should be okay.

At least be sure the checkmanifest trick is in place, which can save your day when making a release later on pyp

**Comment by jalavik** [comment]
Yeah, I will add it (had checked it, but not added to travis)

**Comment by jalavik** [comment]
@kaplun Comments addressed. Thanks!

**Pull request merged**